File: Eunleelee/2024-02/2월 12일/모음사전.py
def solution(word):
    answer = 0
    vowel = ['A', 'E', 'I', 'O', 'U']

    for a in range(5):
        now = vowel[a]
        answer += 1
        if word == now:
          return answer
        for b in range(5):
            now = vowel[a] + vowel[b]
            answer += 1
            if word == now:
              return answer
            for c in range(5):
                now = vowel[a] + vowel[b] + vowel[c]
                answer += 1
                if word == now:
                    return answer
                for d in range(5):
                    now = vowel[a] + vowel[b] + vowel[c] + vowel[d]
                    answer += 1
                    if word == now:
                        return answer
                    for e in range(5):
                        now = vowel[a] + vowel[b] + vowel[c] + vowel[d] + vowel[e]
                        answer += 1
                        if word == now:
                            return answer

    return answer

print(solution("EIO"))

File: yht0827/2월/2월 12일/모음사전.java
class Solution {

	static int answer = 0;
	static int cnt = 0;
	static boolean flag;
	static char[] alpha = {'A', 'E', 'I', 'O', 'U'};

	public int solution(String word) {

		StringBuilder sb = new StringBuilder();
		backTrack(word, sb, 0);

		return answer;
	}

	public void backTrack(String word, StringBuilder sb, int index) {

		if (flag)
			return;

		String s = sb.toString();

		if (word.equals(s)) {
			answer = cnt;
			flag = true;
			return;
		}

		if (index < 5) {
			for (int i = 0; i < 5; i++) {
				sb.append(alpha[i]);
				cnt++;
				backTrack(word, sb, index + 1);
				sb.deleteCharAt(sb.length() - 1);
			}
		}
	}

}